### ISSUE TITLE
ReferenceError: savedJson is not defined

### ISSUE THREAD

**Issue by Ipsis** [irsheep/MMM-SnmpIntSpeed#1]
I've successfully got the correct index configured in the config file, however I see to be unable to retrieve any statistics. (I used the suggested manual call to dump the known interfaces).
I've used SNMP quite a lot, so I'm interested in using this particular module.

I'm getting the following exception in the logs.
> 
0|mm  | Whoops! There was an uncaught exception...
0|mm  | ReferenceError: savedJson is not defined
0|mm  |     at PrepareData (/home/pi/MagicMirror/modules/MMM-SnmpIntSpeed/node_helper.js:113:3)
0|mm  |     at Session.<anonymous> (/home/pi/MagicMirror/modules/MMM-SnmpIntSpeed/node_helper.js:156:54)

I don't see any data/snmp.json file created in the MMM-SnmpIntSpeed directory.
Creating an empty one does allow the code to progress, but then I get failures due to the empty snmp.json being invalid. I've also tried crafting the snmp.json to initialise it with some values, but I've been unable to get the correct format, as I though if I primed the file with some data, it might get past the exception.

Am I just missing something really obvious?

**Comment by irsheep** [comment]
Hi Ipsis,

Did you create the snmp.json inside the data directory "data/snmp.json" ?

Can you tell me what is the device, or if exports the OID ifHCInOctets (1.3.6.1.2.1.31.1.1.1.6)

**Comment by Ipsis** [comment]
I tried creating an empty data/snmp.json and a data/snmp.json with just {}.
The empty one failed with "Cannot read property "19.168.1.11" which is the IP of my device.
The one with {} failed with "Cannot read property "2" which was the snmp index of the interface.

I've tried with a few devices.
A Unifi USG, Unifi WAP and Unifi 8 port switch.
I wrote (after quickly learning some JavaScript) a small script to just dump the relevant oids from the devices. They did respond to snmp requests and that did include  ifHCInOctets.
(I'll update this with their output once I move the code over from my other machine).

My next step was to try and work out what the snmp.json would look like if it did get written, because I'm wondering if that will be enough to kickstart the calculation and rewrite of the file.

(This is all just running on a Raspbian install with MagicMirror and Node v10.15.2)

**Comment by Ipsis** [comment]
A quick update since I was playing with code.
I did manage to get it working by manually creating a valid snmp.json file.

eg.

`
{"192.168.1.1":{"2":{"timestamp":"2019-09-20T20:05:57.644Z","ifDescr":"eth0","ifHighSpeed":10000,"ifHCInOctets":33456,"ifHCOutOctets":99456,"lastbw":{"inBits":56000,"outBits":99123,"ifspeed":1000}}}}
`

Then I restarted MagicMirror with the module enabled for 192.168.1.1 interface 2.
It then successfully parsed the snmp.json and started to populate it with the correct data and update the widget.

So for some reason the module doesn't like starting with a missing or empty snmp.json which it will try to do when it is installed. If you fake the data in the snmp.json for the host/index you are interested in then it seems to get started and recover.

eg.
`
0|mm  | out: 14312      in: 1334488     delta: 26.75    liveTimestamp: 1569009782957    savedTimeStamp: 1569009756207
0|mm  | out: 12928      in: 154912      delta: 8.617    liveTimestamp: 1569009791574    savedTimeStamp: 1569009782957
0|mm  | out: 14568      in: 928176      delta: 5.014    liveTimestamp: 1569009796588    savedTimeStamp: 1569009791574
0|mm  | out: 27816      in: 1614160     delta: 9.999    liveTimestamp: 1569009806587    savedTimeStamp: 1569009796588
0|mm  | out: 2424       in: 3320        delta: 5.001    liveTimestamp: 1569009811588    savedTimeStamp: 1569009806587
0|mm  | out: 1024       in: 3496        delta: 5.041    liveTimestamp: 1569009816629    savedTimeStamp: 1569009811588
0|mm  | out: 1280       in: 3384        delta: 10.041   liveTimestamp: 1569009826670    savedTimeStamp: 1569009816629
0|mm  | out: 3744       in: 4560        delta: 9.961    liveTimestamp: 1569009836631    savedTimeStamp: 1569009826670
0|mm  | out: 1656       in: 1304        delta: 5.001    liveTimestamp: 1569009841632    savedTimeStamp: 1569009836631
0|mm  | out: 16720      in: 20072       delta: 5.029    liveTimestamp: 1569009846661    savedTimeStamp: 1569009841632
0|mm  | out: 7568       in: 97008       delta: 9.971    liveTimestamp: 1569009856632    savedTimeStamp: 1569009846661
0|mm  | out: 34456      in: 402616      delta: 10.108   liveTimestamp: 1569009866740    savedTimeStamp: 1569009856632
0|mm  | out: 269448     in: 36252784    delta: 5.103    liveTimestamp: 1569009871843    savedTimeStamp: 1569009866740
0|mm  | out: 260992     in: 41553896    delta: 9.762    liveTimestamp: 1569009881605    savedTimeStamp: 1569009871843
0|mm  | out: 26813672   in: 697912      delta: 5.038    liveTimestamp: 1569009886643    savedTimeStamp: 1569009881605
`
